src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -0,0 +1,361 @@
+        ].index
+        high_sbp = measured_sbp[measured_sbp >= data_values.SBP_THRESHOLD.HIGH].index
+
+        # Low-SBP patients: do nothing
+
+        # Mid-SBP patients
+        # If tx prescribed, it must be one_drug_half_dose
+        to_prescribe_mid_sbp = mid_sbp.intersection(overcome_therapeutic_inertia)
+        pop_not_medicated.loc[
+            to_prescribe_mid_sbp, self.sbp_medication_column
+        ] = data_values.MEDICATION_RAMP["sbp"][
+            data_values.SBP_MEDICATION_LEVEL.ONE_DRUG_HALF_DOSE
+        ]
+
+        # High-SBP patients
+        # If tx prescribed, apply ramp
+        to_prescribe_high_sbp = high_sbp.intersection(overcome_therapeutic_inertia)
+        pop_not_medicated.loc[
+            to_prescribe_high_sbp, self.sbp_medication_column
+        ] = self.randomness.choice(
+            to_prescribe_high_sbp,
+            choices=list(
+                data_values.FIRST_PRESCRIPTION_LEVEL_PROBABILITY["sbp"]["high"].keys()
+            ),
+            p=list(data_values.FIRST_PRESCRIPTION_LEVEL_PROBABILITY["sbp"]["high"].values()),
+        )
+
+        return pop_not_medicated
+
+    def treat_currently_medicated_sbp(
+        self, pop_medicated: pd.DataFrame, measured_sbp: pd.Series
+    ) -> pd.DataFrame:
+        """Applies the SBP treatment ramp to simulants already on medication
+
+        Arguments:
+            pop_medicated: dataframe of simulants subset to those visiting the doctor and
+                who are already on medication
+            measured_sbp: measured blood pressure values
+        """
+
+        # Generate indexes for simulants who overcome therapeutic inertia, are
+        # adherent to their medication, who are not already medicated, and who
+        # have low and high (measured) SBP levels
+        high_sbp = measured_sbp[measured_sbp >= data_values.SBP_THRESHOLD.HIGH].index
+        overcome_therapeutic_inertia = pop_medicated.index[
+            self.randomness.get_draw(pop_medicated.index)
+            > data_values.THERAPEUTIC_INERTIA_NO_START
+        ]
+        adherent = pop_medicated[
+            pop_medicated[self.sbp_medication_adherence_type_column]
+            == data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        not_already_max_medicated = pop_medicated[
+            pop_medicated[self.sbp_medication_column]
+            < max(data_values.MEDICATION_RAMP["sbp"].values())
+        ].index
+
+        # High-SBP patients
+        # If tx changed, move up ramp
+        medication_change = (
+            high_sbp.intersection(overcome_therapeutic_inertia)
+            .intersection(adherent)
+            .intersection(not_already_max_medicated)
+        )
+        pop_medicated.loc[medication_change, self.sbp_medication_column] = (
+            pop_medicated[self.sbp_medication_column] + 1
+        )
+
+        return pop_medicated
+
+    def apply_sbp_treatment_ramp(self, pop_visitors: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Nit: code is easier to read if you define a function closer to its actual usage. In this case it would be easier to read if `apply_sbp_treatment_ramp` was defined above the `treat_...` methods